### ISSUE TITLE
Septerra Core screen shake when loading new area

### ISSUE THREAD

**Issue by SirPrimalform** [M-HT/SR#53]
I'm not sure if this is something unique to my system, or if I have something configured wrong. When the game is loading a new area (i.e. while the loading bar fills) the last image of the current area shakes extremely fast (alternating between two seemingly non-consecutive frames). This only seems to happen on the SR version, if I run the original exe with dgvoodoo there's no screen shake.

Happy to provide any info I can if told what to look for.

**Comment by Gaerthal-Estelmist** [comment]
I have the exact issue with SirPrimalform. Thankfully it does not break the game, so I'm OK with it. 
In addition to this, I want to ask a somewhat unrelated question. Sorry if it is the wrong place.
The Unofficial SC patch comes with the ability to right click during battle to choose a character as soon as its current endurance bar section fills up. I've tried many things but I just can't make it work with this SR fix. Is there maybe any way to include that mechanic in SR? Thanks in advance.

**Comment by M-HT** [comment]
The screen shaking is how it's implemented currently. I'll take a look at it, but no guarantees.

You can't use the unofficial patch with this version. I'll check if it can be implemented here.

**Comment by Gaerthal-Estelmist** [comment]
Seeing the commit, I guess you are working on this?
As I wrote in my previous message, the Unofficial patch seems to come installed with default gog version. In case you need to have a closer look at the patch itself, here is the forum link and there is an external link to the patch page at the end of the first comment: [https://www.gog.com/forum/septerra_core_legacy_of_the_creator/unofficial_patch/page1](https://www.gog.com/forum/septerra_core_legacy_of_the_creator/unofficial_patch/page1)

I've been checking this page everyday for the possibility that the right-click mechanic of the Unofficial Patch would be implemented in your SR fix. Fingers crossed :)

**Comment by M-HT** [comment]
@Gaerthal-Estelmist I added the option of using right mouse click to preselect characters in combat. It's enabled by default, but can be disabled in the config file.
I'll make a new release soon.

**Comment by M-HT** [comment]
I made a new release, which contains the changes.

**Comment by Gaerthal-Estelmist** [comment]
Hello, first of all thank you for the update. I tested the right-click mechanic and it works great!
Unfortunately, some other problems appeared for me which were not there in the previous version.

As I did in the version before this, I changed the following in the cfg file:
Display_Mode=fullscreen
Display_VSync=on
Display_Width=1440
Display_Height=1080

Option_DefaultMovement=run

CPU_SleepMode=nosleep

Then I started the game. There are two problems and both of them are related to how the cursor behaves:
1- I have 1920x1080 laptop monitor. With the resolutions I chose above, there are black bars to either sides of the screen, as expected (I like it better than stretched images). In the previous version, the cursor used to stop at the border of these black bars and the cursor's tip would not leave the actual game screen, but now it does not stop and move further and disappear completely. When it disappears, I need to drag the mouse for a while for it to reappear in the actual game screen. 
2- The cursor moves smoothly, but WAY slower, as if the DPI was modified nearly all the way down. You know, when you click the DPI mouse on the button and it changes how fast the cursor goes. In the previous version, it was nearly identical with my desktop DPI but now, it moves too slowly as if the DPI button is pressed. Not sure if this is related to problem #1, but I suspect so.

Is there something I can change to fix these behaviours?

**Comment by M-HT** [comment]
I fixed the problem with cursor moving outside the game area.

I changed the way the mouse cursor position is handled, which means the cursor moves slower, but it should move the same as in desktop. Does it work the same if you use `Display_Mode=desktop` in the configuration file ?
What operating system are you using and what is your DPI setting ?

**Comment by Gaerthal-Estelmist** [comment]
Hey, I tested around a little bit as per your questions.
Desktop and fullscreen modes seem to act the same for me as far as mouse DPI is concerned.
I use Windows 8.1 . My mouse is A4-Tech (XL-750BH) which has 600-800-1200-1600-2000-3600 DPI options.
I use the 2000 DPI setting of the mouse on my desktop. Now here are my tests with versions 0.7 and 0.8:
-> In Version 0.7, I need to move the mouse a little (let's say 1 cm for ease of comparing) for it to go from left side to the right side of the game screen. 
-> In Version 0.8, I need to move the mouse more (around 2 cm) for it to go from left side to the right side of the game screen. If I increase the DPI to its max setting of the mouse (which is 3600), I get the nearly exact cursor movement speed of Version 0.7, which takes about 1 cm mouse movement w 2000 DPI to navigate whole game screen.

-> In Version 8, alt-tabbing repeatedly showed me that the mouse movement is more accurate in terms of pixel calculation.  It seems in v0.7, the cursor moves faster in the game than it does on desktop- again seen with repeated alt-tabbing.

These tests showed me that v0.8 is more accurate in terms of pixel calculation and movement. Since the mouse was moving faster in v0.7, I guess I got used to it and when it slowed down in v0.8 , it felt like a problem to me. But in the end, this is not a critical issue, I can always use the mouse's 3600 DPI setting to play the game if it feels too slow.

Can't wait to try your fix. Thanks for your time.
